src/jestUtils.ts

```diff
@@ -172,10 +172,12 @@ function fillOldStateChanges(
     } as GestureHandlerTestEvent;
   }
 
-  const isGestureStateEvent = previousEvent.state !== currentEvent.state;
+  // At this point previousEvent is not null, but for some reason it breaks CI
+  // optional chaining used only to fix it
+  const isGestureStateEvent = previousEvent?.state !== currentEvent?.state;
```

Comment:
Try upgrading TS to 4.5 in the example app (same as in the main package). It may resolve this issue.

Reply:
Done in https://github.com/software-mansion/react-native-gesture-handler/pull/1882/commits/ec3ee214654873c6c9304db755dc31f2b6afd648. Seems like it worked.